compiler/ballerina-treegen/src/main/resources/treegen_config.properties

```diff
@@ -13,3 +13,4 @@ external.node.transformer.template=external_node_transformer_template.mustache
 external.tree.modifier.template=external_tree_modifier_template.mustache
 external.node.factory.template=external_node_factory_template.mustache
 external.node.package=io.ballerina.compiler.syntax.tree
+template.config.data=template_config_data.json
```

Comment:
Regarding the file name:
`syntax_tree_descriptor.json` - keeps the descriptive information related to the syntax tree.
`template_config_data.json` - keeps the additional info like create year, released version w.r.t. a syntax tree node.

Based on the usage, I would suggest a name like `syntax_node_metadata.json`. WDYT?
@ushirask @rdulmina

Reply:
+1